### ISSUE TITLE
Interop default imports are undefined because of function hoisting.

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/parcel#5557]
# 🐛 bug report

In some cases, interop default imports are undefined because of function hoisting.

## 🤔 Expected Behavior

Prints "y".

## 😯 Current Behavior

Prints undefined

```js
(function () {
  function $parcel$interopDefault(a) {
    return a && a.__esModule ? a.default : a;
  }

  // ASSET: y.js
  var $y$exports = {};
  $y$exports = "y";
  $x$var$foo(); // calls `foo` before `$y$$interop$default` was set.
  var $y$$interop$default = $parcel$interopDefault($y$exports);

  function $x$var$foo() {
    console.log($y$$interop$default); // undefined when called
  }
})();

```

## 💁 Possible Solution

Do something similar to https://github.com/parcel-bundler/parcel/pull/5519

## 🔦 Context

Occurs in async-to-generator.

## 💻 Code Sample

```js
// x.js
import y from "./y.js";
foo();
function foo() {
	console.log(y);
}

// y.js
module.exports = "y";
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | aff1426c88f465cdea1a977c4b6633417e61eb69

**Comment by devongovett** [comment]
Should be fixed in #4815

**Comment by danieltroger** [comment]
Seems to have been fixed with the merging of #4815, you can close this issue @mischnic.

Btw thank you so much for aknowledging and promptly fixing issues. I just reported 3 issues in firefox and got told the feature that is buggy doesn't exist, even though there is documentation for it…

As a thank you I will report another parcel issue with too aggressive caching of files from imported modules today if it still persists and I can create a reproduction "repo" lol
